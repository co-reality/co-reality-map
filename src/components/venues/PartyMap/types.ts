export type EventData = {
  start_minute: number;
  duration_minutes: number;
  host: string;
  name: string;
  text: string;
  interactivity: string;
}[];

export interface PartyMapData {
  host: {
    url: string;
    icon: string;
    name: string;
  };
  description: {
    text: string;
    program_url?: string;
  };
  start_utc_seconds: number;
  duration_hours: number;
  entrance_hosted_hours: number;
  party_name: string;
  unhosted_entry_video_url: string;
  map_url: string;
  map_viewbox: string;
  password: string;
  admin_password?: string;
  rooms: RoomData[];
}

export interface RoomData {
  name?: string;
  title: string;
  subtitle: string;
  about: string;
  image: string;
  on_list: boolean;
  template?: string;
  on_map: boolean;
  button_text?: string;
  attendance_x?: string;
  attendance_y?: string;
  url: string;
  path: string;
  events: EventData;
  external_url: string;
}
