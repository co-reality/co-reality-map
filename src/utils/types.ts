export const isTruthy = <T>(item: T): boolean => !!item;

export const hasElements = <T>(array?: T[] | null): boolean =>
  isTruthy(array?.length);

export const notEmpty = <T>(value: T | null | undefined): value is T =>
  value !== null && value !== undefined;
