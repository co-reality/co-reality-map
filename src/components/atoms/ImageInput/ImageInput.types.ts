import { FieldError } from "react-hook-form";

export interface ImageInputProps {
  onChange: (url: string) => void;
  name: string;
  customClass?: string;
  imgUrl?: string;
  error?: FieldError;
  small?: boolean;
  forwardRef: any;
  nameWithUnderscore?: boolean;
}
