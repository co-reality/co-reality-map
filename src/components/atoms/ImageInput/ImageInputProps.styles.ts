import styled, { css } from "styled-components";

type UploadButtonType = {
  isHidden?: boolean;
};
const defaultButton = css`
  background-color: #462a87;

  cursor: pointer;

  &:hover,
  &:active,
  &:focus {
    transform: translateY(-1px);
    background-color: rgba(55, 90, 127, 0.18);
  }
`;
const hiddenButton = css`
  background-color: transparent;
  color: transparent;
  cursor: default;
`;
export const UploadButton = styled.span`
  ${(props: UploadButtonType) =>
    props.isHidden ? hiddenButton : defaultButton};

  padding: 0.7em 4.2em;
  margin: 0;

  border-radius: 22px;

  font-weight: bold;

  transition: transform 0.2s ease-in-out, background-color 0.2s ease-in-out;
`;

type WrapperType = {
  small?: boolean;
  hasError?: boolean;
  backgroundImage?: string;
};

const smallWrapper = css`
  display: inline-block;
  padding: 2.5em;

  ${UploadButton} {
    padding: 0.7em 2em;
  }
`;
const defaultWrapper = css`
  display: flex;
  padding: 2.5em 0;
`;
export const Wrapper = styled.div`
  ${(props: WrapperType) => (props.small ? smallWrapper : defaultWrapper)};

  align-items: center;
  justify-content: center;

  border: ${(props) => (props.hasError ? "2px solid red" : "none")};
  border-radius: 22px;

  background-size: cover;
  background-color: #1a1d24;
  background-image: ${(props) =>
    props.backgroundImage ? `url(${props.backgroundImage})` : "none"};
  background-position: center;

  cursor: pointer;
`;

export const Error = styled.span`
  display: block;

  color: red;
  font-style: italic;
`;
