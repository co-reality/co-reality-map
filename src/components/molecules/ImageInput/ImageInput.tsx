import React, { useMemo } from "react";
import { FieldError } from "react-hook-form";
import "firebase/functions";

interface ImageInputProps {
  disabled: boolean;
  name: string;
  image?: FileList;
  containerClassName?: string;
  imageClassName?: string;
  error?: FieldError;
}

// eslint-disable-next-line
export const ImageInput = React.forwardRef<HTMLInputElement, ImageInputProps>(
  (props, ref) => {
    const {
      image,
      containerClassName,
      imageClassName,
      name,
      error,
      disabled,
    } = props;

    const imageUrl = useMemo(
      () => image && image.length > 0 && URL.createObjectURL(image[0]),
      [image]
    );

    return (
      <>
        <div className={`image-input default-container ${containerClassName}`}>
          {imageUrl ? (
            <img
              className={`default-image ${imageClassName}`}
              src={imageUrl}
              alt="upload"
            />
          ) : (
            <div className="centered-flex empty">
              <h6 className=" text">Click to upload an image</h6>
            </div>
          )}
          <input
            disabled={disabled}
            name={name}
            type="file"
            ref={ref}
            accept="image/x-png,image/gif,image/jpeg"
            className="default-input"
          />
        </div>
        {error?.message && <span className="input-error">{error.message}</span>}
      </>
    );
  }
);
