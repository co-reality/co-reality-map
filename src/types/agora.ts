import {
  IAgoraRTCClient,
  IAgoraRTCRemoteUser,
  ILocalVideoTrack,
} from "agora-rtc-sdk-ng";

export enum AgoraClientConnectionState {
  CONNECTED = "CONNECTED",
  DISCONNECTED = "DISCONNECTED",
  CONNECTING = "CONNECTING",
  RECONNECTING = "RECONNECTING",
  DISCONNECTING = "DISCONNECTING",
}

export interface UseAgoraRemotesProps {
  client: IAgoraRTCClient;
}
export type UseAgoraRemotesReturn = IAgoraRTCRemoteUser[];

export interface UseAgoraScreenShareProps {
  client: IAgoraRTCClient;
}
export interface UseAgoraScreenShareReturn {
  localScreenTrack?: ILocalVideoTrack;
  stopShare(): void;
  shareScreen(): Promise<void>;
  joinChannel(): Promise<void>;
  leaveChannel(): Promise<void>;
}

export interface UseAgoraCameraProps {
  client: IAgoraRTCClient;
}
export interface UseAgoraCameraReturn {
  isCameraOn: boolean;
  isMicrophoneOn: boolean;
  localCameraTrack?: ILocalVideoTrack;
  toggleCamera(): void;
  toggleMicrophone(): void;
  joinChannel(): Promise<void>;
  leaveChannel(): Promise<void>;
}
