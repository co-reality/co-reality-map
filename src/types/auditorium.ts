export type AuditoriumSection = {
  isLocked?: boolean;
  rowsCount?: number;
  columnsCount?: number;
  capacity?: number;
};

export enum AuditoriumSize {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large",
}
