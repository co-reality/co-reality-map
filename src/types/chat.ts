import { IconDefinition } from "@fortawesome/free-solid-svg-icons";
import { faPoll, faQuestion } from "@fortawesome/free-solid-svg-icons";
import { User } from "types/User";
import { WithId } from "utils/id";
import { isTruthy } from "utils/types";
import firebase from "firebase/app";

export type BaseChatMessage = {
  from: string;
  text: string;
  ts_utc: firebase.firestore.Timestamp;
  deleted?: boolean;
  threadId?: string;
  isQuestion?: boolean;
};

export type PrivateChatMessage = BaseChatMessage & {
  to: string;
  isRead?: boolean;
};

export type VenueChatMessage = BaseChatMessage;

export type PollMessage = BaseChatMessage & {
  poll: PollValues;
  votes: string[];
};

export type ChatMessage = PrivateChatMessage | VenueChatMessage | PollMessage;

export const isPollMessage = (r: unknown): r is PollMessage =>
  typeof r === "object" && isTruthy(r) && r.hasOwnProperty("poll");

export type BaseMessageToDisplay<T extends ChatMessage = ChatMessage> = T & {
  author: WithId<User>;
  isMine: boolean;
  canBeDeleted?: boolean;
};

export type MessageToDisplay<
  T extends ChatMessage = ChatMessage
> = BaseMessageToDisplay<T> & {
  replies: WithId<BaseMessageToDisplay<T>>[];
};

export interface SendMesssageProps {
  text: string;
  isQuestion: boolean;
}

export type SendMesssage = (
  message: SendMesssageProps
) => Promise<void> | undefined;

export type DeleteMessage = (messageId: string) => Promise<void> | undefined;

export interface SendChatReplyProps {
  replyText: string;
  threadId: string;
}

export type SendChatReply = (
  props: SendChatReplyProps
) => Promise<void> | undefined;

export type PreviewChatMessage = PrivateChatMessage & {
  counterPartyUser: WithId<User>;
};

export type PreviewChatMessageMap = { [key: string]: PreviewChatMessage };

export type PreviewChatMessageToDisplay = PreviewChatMessage & {
  isMine: boolean;
};

export enum ChatTypes {
  WORLD_CHAT = "WORLD_CHAT",
  VENUE_CHAT = "VENUE_CHAT",
  PRIVATE_CHAT = "PRIVATE_CHAT",
}

export type PrivateChatSettings = {
  openedChatType: ChatTypes.PRIVATE_CHAT;
  recipientId?: string;
};

export type VenueChatSettings = {
  openedChatType: ChatTypes.VENUE_CHAT;
};

export type ChatSettings = PrivateChatSettings | VenueChatSettings;

// @debt Remove it when UserProfileModal is refactored
export type SetSelectedProfile = (user: WithId<User>) => void;

export enum ChatOptionType {
  poll = "poll",
  question = "question",
}

export interface ChatOption {
  icon: IconDefinition;
  name: string;
}

export type ChatOptionMap = Record<ChatOptionType, ChatOption>;

export type PollQuestion = {
  name: string;
  id?: number;
  votes?: number;
};

export type PollValues = {
  topic: string;
  questions: PollQuestion[];
};

export const ChatMessageOptions: ChatOptionMap = {
  poll: {
    icon: faPoll,
    name: "Create Poll",
  },
  question: {
    icon: faQuestion,
    name: "Ask Question",
  },
};
