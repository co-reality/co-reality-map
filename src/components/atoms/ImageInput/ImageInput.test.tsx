import React from "react";
import renderer from "react-test-renderer";

// Test component
import ImageInput from "./ImageInput";

// Typings
import { ImageInputProps } from "./ImageInput.types";

describe("<ImageInput />", () => {
  test("renders correctly", () => {
    const ref = React.createRef();
    const props = {
      name: "imageInputName",
      onChange: () => {},
      forwardRef: ref,
    } as ImageInputProps;

    const tree = renderer.create(<ImageInput {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
