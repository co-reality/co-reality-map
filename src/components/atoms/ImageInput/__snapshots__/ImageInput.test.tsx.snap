// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageInput /> renders correctly 1`] = `
Array [
  <label
    className="sc-gsTCUz gGtXOY"
  >
    <input
      accept="image/x-png,image/gif,image/jpeg"
      className=""
      hidden={true}
      id="imageInputName"
      name="imageInputNameFile"
      onChange={[Function]}
      type="file"
    />
    <span
      className="sc-bdfBwQ izHeuX"
    >
      Upload
    </span>
  </label>,
  <input
    name="imageInputNameUrl"
    readOnly={true}
    type="hidden"
    value=""
  />,
]
`;
